src/lib/Name.hs

```diff
@@ -2121,55 +2118,81 @@ instance Store (Scope n)
 deriving instance (forall c n. Pretty (v c n)) => Pretty (RecSubstFrag v o o')
 
 
-type EE = EitherE
-
-type EitherE1 e0                = EE e0 VoidE
-type EitherE2 e0 e1             = EE e0 (EE e1 VoidE)
-type EitherE3 e0 e1 e2          = EE e0 (EE e1 (EE e2 VoidE))
-type EitherE4 e0 e1 e2 e3       = EE e0 (EE e1 (EE e2 (EE e3 VoidE)))
-type EitherE5 e0 e1 e2 e3 e4    = EE e0 (EE e1 (EE e2 (EE e3 (EE e4 VoidE))))
-type EitherE6 e0 e1 e2 e3 e4 e5 = EE e0 (EE e1 (EE e2 (EE e3 (EE e4 (EE e5 VoidE)))))
-
-pattern Case0 :: e0 n -> EE e0 rest n
-pattern Case0 e = LeftE e
-
-pattern Case1 :: e1 n -> EE e0 (EE e1 rest) n
-pattern Case1 e = RightE (LeftE e)
-
-pattern Case2 :: e2 n -> EE e0 (EE e1 (EE e2 rest)) n
-pattern Case2 e = RightE (RightE (LeftE e))
-
-pattern Case3 :: e3 n -> EE e0 (EE e1 (EE e2 (EE e3 rest))) n
-pattern Case3 e = RightE (RightE (RightE (LeftE e)))
-
-pattern Case4 :: e4 n ->  EE e0 (EE e1 (EE e2 (EE e3 (EE e4 rest)))) n
-pattern Case4 e = RightE (RightE (RightE (RightE (LeftE e))))
-
-pattern Case5 :: e5 n ->  EE e0 (EE e1 (EE e2 (EE e3 (EE e4 (EE e5 rest))))) n
-pattern Case5 e = RightE (RightE (RightE (RightE (RightE (LeftE e)))))
-
-type EB = EitherB
-
-type EitherB1 e0             = EB e0 VoidB
-type EitherB2 e0 e1          = EB e0 (EB e1 VoidB)
-type EitherB3 e0 e1 e2       = EB e0 (EB e1 (EB e2 VoidB))
-type EitherB4 e0 e1 e2 e3    = EB e0 (EB e1 (EB e2 (EB e3 VoidB)))
-type EitherB5 e0 e1 e2 e3 e4 = EB e0 (EB e1 (EB e2 (EB e3 (EB e4 VoidB))))
-
-pattern CaseB0 :: e0 n l -> EB e0 rest n l
-pattern CaseB0 e = LeftB e
-
-pattern CaseB1 :: e1 n l -> EB e0 (EB e1 rest) n l
-pattern CaseB1 e = RightB (LeftB e)
-
-pattern CaseB2 :: e2 n l -> EB e0 (EB e1 (EB e2 rest)) n l
-pattern CaseB2 e = RightB (RightB (LeftB e))
-
-pattern CaseB3 :: e3 n l -> EB e0 (EB e1 (EB e2 (EB e3 rest))) n l
-pattern CaseB3 e = RightB (RightB (RightB (LeftB e)))
+data EitherE6 (e0::E) (e1::E) (e2::E) (e3::E) (e4::E) (e5::E) (n::S)
```

Comment:
Some (all?) of the explanation you gave for this change in your PR message would go well here as a code comment, explaining why this is 6 and not 5 or 7, and which it's a flat structure rather than a chain of pairs.

Reply:
Added a comment. Some of the details from PR description are more before vs after, and I skipped those. One can always find them through git history.